app/assets/javascripts/features/confirm_dialog.js

```diff
@@ -0,0 +1,32 @@
+$(function(){
+  $(document).on('click', '.confirm-dialog-wrapper .close-dialog', function(ev){
+    $(this).closest('.confirm-dialog-wrapper').fadeOut(100);
+  });
+});
+
+// Override the default confirm dialog by rails
+$.rails.allowAction = function(link){
+  if (link.data("confirm") == undefined){
+    return true;
+  }
+  $.rails.showConfirmationDialog(link);
+  return false;
+}
+
+// Display the confirmation dialog
+$.rails.showConfirmationDialog = function(link){
+  var message = link.data("confirm");
```

Comment:
How do you mean? Is it not always going to need the data-confirm as a message anyway? We could maybe have an optional field for updating the text on the confirm button itself?